include/proto/functions.proto

```diff
@@ -86,17 +92,29 @@ message DagSchedule {
   required string response_id = 2;
   required Dag dag = 3;
   required string target_function = 4;
-  repeated string triggers = 5;
-  map<string, string> locations = 6; // which functions should be executed where
+  required ConsistencyType consistency = 5 [default = NORMAL]; 
+  repeated string triggers = 6;
+  map<string, string> locations = 7; // which functions should be executed where
   // the pre-fixed arguments for each stage (i.e., the arguments that do not depend on the previous stage)
-  map<string, ArgList> arguments = 7; 
+  map<string, ArgList> arguments = 8; 
+}
+
+message VersionedKey {
+  required string key = 1;
+  map<string, uint32> vector_clock = 2;
+}
+
+message VersionedKeyList { 
+  repeated VersionedKey versioned_keys = 1;
 }
 
 message DagTrigger {
   required uint64 id = 1;
   required string target_function = 2; // which function is this trigger for
   required string source = 3;
   optional ArgList arguments = 4;
+  map<string, VersionedKeyList> address_to_versioned_key_list_map = 5;
```

Comment:
Can we title this `versioned_key_locations`? Much less cumbersome to read.

Reply:
yes sir